src/store/authentication/saga.ts

```diff
@@ -108,7 +115,13 @@ export function* initializeUserState(user: User) {
 }
 
 export function* clearUserState() {
-  yield clearUserLayout();
+  yield all([
+    call(clearChannelsAndConversations),
+    call(clearMessages),
+    call(clearUsers),
+    call(clearNotifications),
+    call(clearUserLayout),
+  ]);
```

Comment:
I think this is a great place where using saga channels will be very useful. This is fine for now but let's keep that in mind when we're in here next time.